caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java

```diff
@@ -321,4 +306,21 @@ public URL toURL(int serialNum, URI uri, Cutout cutout, Map<String, List<String>
         }
         return ret;
     }
+    
+    private URL errorURL(String errorMessage) throws MalformedURLException {
+        StringBuilder path = new StringBuilder();
+        path.append("400");
+        path.append("|text/plain");
+        path.append("|").append(errorMessage);
+        String msg = Base64.encodeString(path.toString());
+
+        // hack to get base url for soda service
+        URL serviceURL = reg.getServiceURL(sodaURI, Standards.SODA_SYNC_10, AuthMethod.ANON);
+        URL url = new URL(serviceURL.toExternalForm() + "/" + msg);
+        String surl = serviceURL.toExternalForm().replace("/sync", "/soda-echo");
+        surl = surl + "/" + msg;
+        URL loc = new URL(url.toExternalForm().replace("/sync", "/soda-echo"));
```

Comment:
this `new URL` ignores the `surl` put together above

Reply:
Fixed.